### ISSUE TITLE
How to create partial_double of WebSocketClient?

### ISSUE THREAD

**Issue by token-cjg** [bitwes/Gut#368]
Thanks for writing this marvelous testing library!

**GUT version**: 7.3.0
**Godot version**: 3.4.3 (MacOS)

### Description

According to https://github.com/bitwes/Gut/wiki/Doubles#doubling-built-ins, I gather that one can do things like this:

```
var partial_doubled_raycast = partial_double(Raycast2D).new()
```
and then stub / mock methods accordingly.  Indeed if I write a test with GUT 7.3.0 this all works.

However, if I try this for [WebSocketClient in Godot 3.4.3](https://docs.godotengine.org/en/3.4/classes/class_websocketclient.html)

```
var partial_doubled_websocketclient = partial_double(WebSocketClient).new()
```

and then try running the GUT test harness, the build crashes.

### Hypothesis

I understand that RayCast2D inherits from [Node](https://docs.godotengine.org/en/3.4/classes/class_node.html#class-node), but WebSocketClient inherits from [Reference](https://docs.godotengine.org/en/3.4/classes/class_reference.html#class-reference).

Is it the case that only built-ins that inherit from Node can be mocked at present?

**Comment by bitwes** [comment]
This appears to be a bug.  The doubler is having issues the two parameters with `PoolStringArray` defaults in `connect_url`:
```
 Error connect_to_url(url: String, protocols: PoolStringArray = PoolStringArray(  ), gd_mp_api: bool = false, custom_headers: PoolStringArray = PoolStringArray(  ))
```

I tried all the tricks I could think of to get around this.  I tried stubbing the `param_count` and `param_defaults`.  I also tried using `ignore_method_when_doubling`.  I believe the issue with these approaches is that they are only compatible with scripts and scenes.


## The Error
```
Debugger Break, Reason: 'Parser Error: Error parsing expression, misplaced: ']''
*Frame 0 - :59 in function ''
Enter "help" for assistance.
debug> 
```

I got the source for the double that was being generated.  This is the problem line:
```
func connect_to_url(p_url=__gut_default_val("connect_to_url",0), p_protocols=Null[], p_gd_mp_api=false, p_custom_headers=Null[]):
```
Which makes sense because `Null[]` is not valid GDScript.


## Hacky Workaround
If you change `_get_arg_text` in `res://addons/gut/method_maker.gd` by adding `text = text.replace('Null[]', '[]')` it looks like it works. 
```
func _get_arg_text(arg_array):
	var text = ''

	for i in range(arg_array.size()):
		text += str(arg_array[i].p_name, '=', arg_array[i].default)
		if(i != arg_array.size() -1):
			text += ', '
	text = text.replace('Null[]', '[]') # <------ add this line
	return text
```

I was able to get the test below to pass after making the change.  I don't know if it will work if you stub the method `to_call_super`, but it should work for spying or if you don't need to test this method's usage.
```
func test_make_websocket_partial_double():
	var WebSocketClientPD = partial_double(WebSocketClient)
	assert_not_null(WebSocketClientPD)
	var pd = WebSocketClientPD.new()
	assert_not_null(pd)
```
Beware, this is a SUPER HACK and I didn't really test it at all.  But it might work until I can figure out how to really solve the issue.

_Note to self, add `gut.get_doubler()._print_source = true` in a test to make GUT spit out the source code for a double  I'll forget how to do this again, but at least it's documented here where I'm sure to forget it._

**Comment by bitwes** [comment]
P.S. you are welcome, thanks for using GUT!

**Comment by token-cjg** [comment]
Thanks for the quick response!

**Comment by bitwes** [comment]
Released in 7.4.0
